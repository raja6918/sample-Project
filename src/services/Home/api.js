import apiUtils from '../../utils/API/utils';

const homeAPI = apiUtils.buildAPI(apiUtils.getConfig().HOME_API);

export default homeAPI;
