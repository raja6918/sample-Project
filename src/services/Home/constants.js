export const HOME_ENDPOINT = '/data/data-home';
