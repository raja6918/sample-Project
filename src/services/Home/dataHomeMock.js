const dataHomeMock = {
  scenarioId: 1002,
  binId: 1,
  dataTypes: [
    {
      type: 'accommodation',
      source: 'import',
      count: 65,
      currentFile: {
        version: 2,
        binId: 1,
        binName: 'Default',
      },
      availableVersions: [1, 2],
      errorCount: null,
      newFilesAvailable: [
        {
          id: 2,
          version: 2,
          userDisplayName: 'Brandon Padilla',
          creationTime: '2019-11-11T10:15:53.630Z',
        },
      ],
    },
    {
      type: 'aircraftType',
      source: 'template',
      count: 3,
      currentFile: null,
      availableVersions: null,
      errorCount: 5,
      newFilesAvailable: [],
    },
    {
      type: 'commercialFlight',
      source: 'import',
      count: null,
      errorCount: null,
      currentFile: {
        version: 1,
        binId: 1,
        binName: 'Default',
      },
      availableVersions: [1, 2, 3, 4, 5],
      newFilesAvailable: [
        {
          id: 4,
          version: 2,
          userDisplayName: 'Brandon Padilla',
          creationTime: '2019-11-11T10:15:53.630Z',
        },
        {
          id: 5,
          version: 3,
          userDisplayName: 'Jonathan Sanchez',
          creationTime: '2019-09-05T13:10:53.630Z',
        },
        {
          id: 6,
          version: 4,
          userDisplayName: 'Alberto Chac√≥n',
          creationTime: '2019-10-06T17:22:53.630Z',
        },
        {
          id: 7,
          version: 5,
          userDisplayName: 'Christian Portillo',
          creationTime: '2019-11-01T09:22:53.630Z',
        },
        {
          id: 15,
          version: 6,
          userDisplayName: 'David Morales',
          creationTime: '2019-10-31T11:45:53.630Z',
        },
      ],
    },
    {
      type: 'coterminalTransport',
      source: '',
      count: 4,
      currentFile: null,
      availableVersions: null,
      errorCount: null,
      newFilesAvailable: [],
    },
    {
      type: 'country',
      count: 248,
      currentFile: null,
      availableVersions: [1, 2, 3],
      errorCount: null,
      newFilesAvailable: [],
    },
    {
      type: 'base',
      count: 4,
      currentFile: null,
      availableVersions: null,
      errorCount: null,
      newFilesAvailable: [],
    },
    {
      type: 'currency',
      source: '',
      count: 157,
      currentFile: null,
      availableVersions: null,
      errorCount: null,
      newFilesAvailable: [],
    },
    {
      type: 'flight',
      source: 'import',
      count: 497,
      errorCount: null,
      currentFile: {
        version: 3,
        binId: 1,
        binName: 'Default',
      },
      availableVersions: [1, 2, 3, 4],
      newFilesAvailable: [],
    },
    {
      type: 'position',
      source: 'template',
      count: 4,
      currentFile: null,
      availableVersions: null,
      errorCount: null,
      newFilesAvailable: [],
    },
    {
      type: 'region',
      source: '',
      count: 5,
      currentFile: null,
      availableVersions: null,
      errorCount: null,
      newFilesAvailable: [],
    },
    {
      type: 'rule',
      source: 'template',
      count: null,
      currentFile: null,
      availableVersions: null,
      errorCount: null,
      newFilesAvailable: [],
    },
    {
      type: 'station',
      source: 'template',
      count: 111,
      currentFile: null,
      availableVersions: null,
      errorCount: null,
      newFilesAvailable: [],
    },
    {
      type: 'crewGroup',
      source: 'template',
      count: 8,
      currentFile: null,
      availableVersions: null,
      errorCount: null,
      newFilesAvailable: [],
    },
  ],
};

export default dataHomeMock;
