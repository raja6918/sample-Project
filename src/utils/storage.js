const storageType = window.sessionStorage;
const localStorageType = window.localStorage;

const storage = {
  length: () => storageType.length,
  clear: storageType.clear,
  key: storageType.key,
  removeItem: key => storageType.removeItem(key),
  getItem: key => JSON.parse(storageType.getItem(key)),
  setItem: (key, value) => storageType.setItem(key, JSON.stringify(value)),
};

export const localStorage = {
  length: () => localStorageType.length,
  clear: localStorageType.clear,
  key: localStorageType.key,
  removeItem: key => localStorageType.removeItem(key),
  getItem: key => JSON.parse(localStorageType.getItem(key)),
  setItem: (key, value) => localStorageType.setItem(key, JSON.stringify(value)),
};

export default storage;
