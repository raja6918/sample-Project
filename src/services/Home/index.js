import API from './api';
import { HOME_ENDPOINT } from './constants';

export function getDataHomeCards(scenarioId, userId) {
  const params = `userId=${userId}&scenarioId=${scenarioId}`;
  return API.get(`${HOME_ENDPOINT}?${params}`);
}

export default {
  getDataHomeCards,
};
