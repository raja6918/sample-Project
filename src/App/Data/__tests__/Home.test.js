import React from 'react';
import { shallow } from 'enzyme';

import storage from '../../../utils/storage';

import { Home } from '../Home';

jest.mock('../../../utils/storage');

const minProps = {
  t: jest.fn(),
  openItemName: 'openItem Test',
  path: '/data',
  openItemId: '1001',
  cards: [],
  importProcess: [],
  isFetching: false,
  setIsFetching: () => {},
  userData: { id: 1 },
};
const wrapper = shallow(<Home {...minProps} />);

const mockStorage = {
  openScenario: {
    startDate: Date.now(),
    endDate: Date.now(),
  },
  loggedUser: {
    id: '1',
  },
};

describe('Home Component', () => {
  let mock;

  function clearMock() {
    if (mock) mock.mockReset();
  }
  beforeAll(clearMock);

  afterAll(() => mock.restore());

  beforeEach(() => {
    mock = storage.getItem.mockImplementation(key => mockStorage[key]);
  });

  afterEach(clearMock);

  test('Home component renders correct', () => {
    expect(wrapper).toMatchSnapshot();

    const emptyCells = wrapper.state().emptyCells.length;
    expect(emptyCells).toBe(0);

    wrapper.setState({ emptyCells: [{}, {}] });

    wrapper.instance().forceUpdate();

    wrapper.unmount();
  });

  test('Header name to match with selected openItem name', () => {
    const wrapper = shallow(<Home {...minProps} />);
    expect(
      wrapper
        .find('Connect(AccessEnabler)')
        .at(0)
        .renderProp('render')({ disableComponent: false })
        .find('span')
        .text()
    ).toEqual(minProps.openItemName + ' ');
  });
});
