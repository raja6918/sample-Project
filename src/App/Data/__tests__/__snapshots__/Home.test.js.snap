// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Home component renders correct 1`] = `
<ScrollBar
  className=""
  component="div"
  containerRef={[Function]}
  option={
    Object {
      "swipeEasing": true,
      "wheelSpeed": 0.4,
    }
  }
>
  <Home__MainContainer>
    <Home__Header>
      <h2 />
      <Connect(AccessEnabler)
        disableComponent={true}
        render={[Function]}
        scopes={
          Array [
            "scenario_manage",
            "sc_manage",
          ]
        }
      />
      <Connect(AccessEnabler)
        disableComponent={true}
        render={[Function]}
        scopes={
          Array [
            "import_action",
          ]
        }
      />
    </Home__Header>
    <ScrollBar
      className=""
      component="div"
      containerRef={[Function]}
      option={
        Object {
          "swipeEasing": true,
          "wheelSpeed": 0.4,
        }
      }
    >
      <Home__CardsContainer
        editMode={false}
        importInProgress={null}
        innerRef={[Function]}
      >
        <div />
      </Home__CardsContainer>
    </ScrollBar>
    <ImportDialog
      bins={Array []}
      closeDialog={[Function]}
      connectToBin={[Function]}
      createNewBin={[Function]}
      formId="importForm"
      handleImport={[Function]}
      open={false}
      scenarioBinId={null}
      showNotification={[Function]}
      t={
        [MockFunction] {
          "calls": Array [
            Array [
              "SCENARIOS.viewOnly.viewOnlyText",
            ],
            Array [
              "DATA.dataHome",
            ],
            Array [
              "DATA.import.stopDialogTitle",
            ],
            Array [
              "DATA.import.stopDialogText",
            ],
            Array [
              "DATA.import.stopDialogTitle",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      toggleLoader={[Function]}
    />
    <VersionsDialog
      closeDialog={[Function]}
      dataName=""
      formId="versionsForm"
      handleOk={[Function]}
      open={false}
      scenarioBinId={null}
      t={
        [MockFunction] {
          "calls": Array [
            Array [
              "SCENARIOS.viewOnly.viewOnlyText",
            ],
            Array [
              "DATA.dataHome",
            ],
            Array [
              "DATA.import.stopDialogTitle",
            ],
            Array [
              "DATA.import.stopDialogText",
            ],
            Array [
              "DATA.import.stopDialogTitle",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      versions={Array []}
    />
    <NewDataDialog
      closeDialog={[Function]}
      dataFiles={Array []}
      dataName=""
      formId="newDataForm"
      handleOk={[Function]}
      isConnectedToBin={false}
      open={false}
      scenarioBinId={null}
      t={
        [MockFunction] {
          "calls": Array [
            Array [
              "SCENARIOS.viewOnly.viewOnlyText",
            ],
            Array [
              "DATA.dataHome",
            ],
            Array [
              "DATA.import.stopDialogTitle",
            ],
            Array [
              "DATA.import.stopDialogText",
            ],
            Array [
              "DATA.import.stopDialogTitle",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    />
    <DataConversionDialog
      closeDialog={[Function]}
      conversionWarnings={Object {}}
      formId="DataConversionForm"
      handleOk={[Function]}
      open={false}
      t={
        [MockFunction] {
          "calls": Array [
            Array [
              "SCENARIOS.viewOnly.viewOnlyText",
            ],
            Array [
              "DATA.dataHome",
            ],
            Array [
              "DATA.import.stopDialogTitle",
            ],
            Array [
              "DATA.import.stopDialogText",
            ],
            Array [
              "DATA.import.stopDialogTitle",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    />
    <StopDialog
      formId="DataConversionForm"
      handleClose={[Function]}
      handleOk={[Function]}
      open={false}
      t={
        [MockFunction] {
          "calls": Array [
            Array [
              "SCENARIOS.viewOnly.viewOnlyText",
            ],
            Array [
              "DATA.dataHome",
            ],
            Array [
              "DATA.import.stopDialogTitle",
            ],
            Array [
              "DATA.import.stopDialogText",
            ],
            Array [
              "DATA.import.stopDialogTitle",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      texts={
        Object {
          "content": undefined,
          "okButton": undefined,
          "title": undefined,
        }
      }
    />
    <ImportErrorDialog
      closeDialog={[Function]}
      closeOnly={true}
      conversionErrors={Array []}
      formId="ImportErrorForm"
      isInternalError={false}
      open={false}
      t={
        [MockFunction] {
          "calls": Array [
            Array [
              "SCENARIOS.viewOnly.viewOnlyText",
            ],
            Array [
              "DATA.dataHome",
            ],
            Array [
              "DATA.import.stopDialogTitle",
            ],
            Array [
              "DATA.import.stopDialogText",
            ],
            Array [
              "DATA.import.stopDialogTitle",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    />
    <ModalLoader
      background=""
      color="white"
      loaderRequired={true}
      onClose={[Function]}
      open={false}
      subtitle=""
      title=""
    />
    <Notification
      autoHideDuration={5000}
      bottomheight={0}
      buttonText=""
      clear={[Function]}
      displayCloseButton={false}
      message={null}
      onLinkAction={[Function]}
      renderers={
        Object {
          "paragraph": "span",
        }
      }
      type=""
    />
  </Home__MainContainer>
</ScrollBar>
`;
